src/ServiceControl.MSMQ.DLQMonitor/CheckDeadLetterQueue.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace ServiceControl.MSMQ.DLQMonitor
+{
+    using System;
+    using NServiceBus.CustomChecks;
+    using System.Diagnostics;
+
+    public class CheckDeadLetterQueue : CustomCheck
+    {
+        PerformanceCounter dlqPerformanceCounter;
+
+        public CheckDeadLetterQueue() :
+            base("Dead Letter Queue", "Transport", TimeSpan.FromHours(1))
```

Comment:
Should these be user configurable?

Reply:
We could expose it as a config file setting.

DLQ (for MSMQ at least) will only be populated if a message has been in the outbox for more than 4 days.

I'm not sure about ASB. What would a reasonable default be for that?

Reply:
> What would a reasonable default be for that?

That's a question I was asking [here](https://github.com/Particular/ServiceControl/pull/1043#issuecomment-348859826) @WilliamBZA.

With ASB a message will show up as soon as it gets to DLQ. 1h as a default does sound reasonable. As for configuration - if it's something that would make sense, for both checks and it adds value, then can add. I would still want to understand what added value would be though.

Reply:
I agree 1 hour is a sensible default.

The added value would be to allow the customer to change it if they disagree with our default. We are saying they don't need to know as much as 1 hour after messages end up in the DLQ, a customer may wish to be notified within a shorter timeframe.

I guess it is easy enough to add, so we could wait until a customer asks before creating the option.

Reply:
👍 reading my mind @boblangley 🙂